src/Nowruz/modules/layout/components/linkItem/LinkItem.tsx

```diff
@@ -24,9 +37,9 @@ export const LinkItem: React.FC<LinkItemProps> = (props) => {
             <span className={css.itemLabel} style={{ pointerEvents: 'none' }}>
               {label}
             </span>
-            {badgeIcon && menuOpen ? (
+            {badgeIconVal && menuOpen ? (
```

Comment:
here you can change it to this code snippet (without redundant `useEffect`)

`{children?.length && (
              <div className="mr-0 ml-auto cursor-pointer z-50" style={{ pointerEvents: 'none' }}>
                <Icon
                  name={subMenuOpen ? 'chevron-up' : 'chevron-down'}
                  className="text-Brand-300 !cursor-pointer"
                  fontSize={20}
                />
              </div>
   )}`

Reply:
We need badgeIcon, it is not just for child menu, in some cases we should add a badgeIcon in front of a menu Item,  for example, in messages, later we should display a badge containing unread messages count. That's why I implemented it in this way. We already have commented badge Icon for Nowruz, but soon we should add them again

Reply:
I know, but you can handle it with a condition.
from what you said, I got it some `LinkItem`s may have a badge as prop but others may have a sub-menu 
so you can put this condition:
`
children.length ? ({
<div className="mr-0 ml-auto cursor-pointer z-50" style={{ pointerEvents: 'none' }}> <Icon name={subMenuOpen ? 'chevron-up' : 'chevron-down'} className="text-Brand-300 !cursor-pointer" fontSize={20} /> </div>
}) : badgeIcon
`

if it is not like what I said, please skip it and consider it resolved. (Thanks :>)